pydantic/config.py

```diff
@@ -36,183 +34,159 @@ class Extra(str, Enum):
 
-    # whether `Union` should check all allowed types before even trying to coerce
-    smart_union: bool = False  # TODO remove
     # whether dataclass `__post_init__` should be run before or after validation
-    post_init_call: Literal['before_validation', 'after_validation'] = 'before_validation'  # TODO remove
-
-    @classmethod
-    def get_field_info(cls, name: str) -> Dict[str, Any]:
-        """
-        Get properties of FieldInfo from the `fields` property of the config class.
-        """
-
-        fields_value = cls.fields.get(name)
-
-        if isinstance(fields_value, str):
-            field_info: Dict[str, Any] = {'alias': fields_value}
-        elif isinstance(fields_value, dict):
-            field_info = fields_value
-        else:
-            field_info = {}
-
-        if 'alias' in field_info:
-            field_info.setdefault('alias_priority', 2)
-
-        if field_info.get('alias_priority', 0) <= 1 and cls.alias_generator:
-            alias = cls.alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
-            field_info.update(alias=alias, alias_priority=1)
-        return field_info
-
-    @classmethod
-    def prepare_field(cls, field: Any) -> None:
-        """
-        Optional hook to check or modify fields during model creation.
-        """
-        pass
-
-
-def get_config(config: Union[ConfigDict, Type[object], None]) -> Type[BaseConfig]:
+    post_init_call: Literal['before_validation', 'after_validation']  # TODO remove
+
+    # TODO where to impelement these two classmethods?
+    # @classmethod
+    # def get_field_info(cls, name: str) -> Dict[str, Any]:
+    #     """
+    #     Get properties of FieldInfo from the `fields` property of the config class.
+    #     """
+
+    #     fields_value = cls.fields.get(name)
+
+    #     if isinstance(fields_value, str):
+    #         field_info: Dict[str, Any] = {'alias': fields_value}
+    #     elif isinstance(fields_value, dict):
+    #         field_info = fields_value
+    #     else:
+    #         field_info = {}
+
+    #     if 'alias' in field_info:
+    #         field_info.setdefault('alias_priority', 2)
+
+    #     if field_info.get('alias_priority', 0) <= 1 and cls.alias_generator:
+    #         alias = cls.alias_generator(name)
+    #         if not isinstance(alias, str):
+    #             raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
+    #         field_info.update(alias=alias, alias_priority=1)
+    #     return field_info
+
+    # @classmethod
+    # def prepare_field(cls, field: Any) -> None:
+    #     """
+    #     Optional hook to check or modify fields during model creation.
+    #     """
+    #     pass
+
+
+config_keys = set(ConfigDict.__annotations__.keys())
+
+
+def _default_config() -> ConfigDict:
```

Comment:
This is my main question about this, PR I would argue that we shouldn't build full config for all models/validation

Instead we should have a static `default_config` dict, any then maybe a utility method, like `def get_config_param(config, name)` which falls back to `default_config`.

WDYT?

Reply:
We could even for simplicity have something like

```py
class ConfigWrapper:
    title: Optional[str]
    anystr_lower: bool
    anystr_strip_whitespace: bool
    ...

    __slots__ = ('_config',)
    def __init__(self, config: ConfigDict | None):
        self._config = config
    
    def __getattr__(self, name: str) -> Any:
        # something like...
        if self._config:
            v = self._config.get(name)
            if v is not None:
                return v
        return default_config[name]
        ...
```

I know it would involve some duplication, but it would make usage through much more elegant.

Reply:
We then assign an instance of ConfigWrapper to `model_config` when creating the class? It feels always a bit weird to me changing the value of a variable the user has set and changing the type even more. And if  we do that `model_config` would still have the type `ConfigDict` during type checking, right?

Another question is, do you rather want to write `model_config['extra']` or `model_config.extra` for getting the value.

I was also thinking about something like a `ChainMap`, but that behaves like a `TypedDict` for type checking.

Reply:
I guess you're right, just keep it as a dict/None and use a utility to get the value.